app/src/proxy.ts

```diff
@@ -0,0 +1,108 @@
+import child_process from "node:child_process";
+import { JSONValue } from "./utils";
+
+export type ProxyRequest = {
+  method: "GET" | "POST" | "DELETE";
+  path_query: string;
+  stream: false;
+  body?: string;
+  headers: object;
+};
+
+export type ProxyResponse = {
+  data: JSONValue;
```

Comment:
I believe we can always assume this will be a JSON object (the Python code currently types it as `dict`).